src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private Node<T> getNodeByIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " out of bound, size is " + size);
+        }
+        if (index < size / 2) {
+            Node<T> currentNode = head;
```

Comment:
you can declare this variable out of if-condition and depending on the condition assign it a value to avoid creating redundant variable